### ISSUE TITLE
make configuration key in Consul configurable

### ISSUE THREAD

**Issue by zpf1989** [ThreeMammals/Ocelot#346]
Hi，we are using multi-gateway with ReRoutes in one consul store,and ReRoutes are diffrent for each gateway.
I see Ocelot can only recognize the fixed key InternalConfiguration.
Could you make it configurable？
Thanks for your time!

**Comment by TomPallister** [comment]
@zpf1989 yes we can definitely take a look at this! How do you think this should work?

**Comment by zpf1989** [comment]
Well,should we add a node(something named like ConfigurationKey) for the ServiceDiscoveryProvider node in ocelot.json ,besides the Provider,Host and Port?
And when fetching configuration from Consul,ConsulFileConfigurationRepository may use the  ConfigurationKey first,then the default key(InternalConfiguration) if ConfigurationKey is empty.

**Comment by TomPallister** [comment]
@zpf1989 sounds perfect to me!

**Comment by zpf1989** [comment]
Thank you very much!! 
I will try it later.
